crates/services/gas_price_service/src/lib.rs

```diff
@@ -0,0 +1,234 @@
+use async_trait::async_trait;
+use fuel_core_services::{
+    RunnableService,
+    RunnableTask,
+    ServiceRunner,
+    StateWatcher,
+};
+use fuel_core_types::fuel_types::BlockHeight;
+use std::sync::Arc;
+
+use tokio::sync::{
+    RwLock,
+    RwLockReadGuard,
+};
+
+pub mod static_updater;
+
+pub fn new_service<A, U>(
+    current_fuel_block_height: BlockHeight,
+    update_algo: U,
+) -> anyhow::Result<ServiceRunner<GasPriceService<A, U>>>
+where
+    U: UpdateAlgorithm<Algorithm = A> + Send + Sync,
+    A: Send + Sync,
+{
+    let service = GasPriceService::new(current_fuel_block_height, update_algo);
+    Ok(ServiceRunner::new(service))
+}
+
+/// The service that updates the gas price algorithm.
+pub struct GasPriceService<A, U> {
+    /// The algorithm that can be used in the next block
+    next_block_algorithm: SharedGasPriceAlgo<A>,
+    /// The code that is run to update your specific algorithm
+    update_algorithm: U,
```

Comment:
For me, it looks like the updater and algorithm should be one entity since only the algorithm knows how it works inside and what data it requires.

But we can leave it for now as it is and see how it will look like with a nonstatic algorithm.

Reply:
I'm leaning to this approach so that the consumer of the algo doesn't need to have all the generics for the updater to read the values.